src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.AspNetCore.Components;
+using Microsoft.VisualBasic;
+using MudBlazor.Charts;
+using Nox.Ui.Blazor.Lib.Models;
+
+namespace Nox.Ui.Blazor.Lib.Components.Generic;
+
+public partial class SelectList : ComponentBase
+{
+
+    #region Declarations
+
+    [Parameter]
+    public List<SelectEntityModel> EntityList { get; set; } = new();
+
+    public string? CurrentIdList { get; set; }
+
+    public string? DisplaySelectedList => EntityList.Any() ? string.Join(", ", EntityList.Select(Entity => Entity.Name)) : string.Empty;
+
+    [Parameter]
+    public string? Title { get; set; }
+
+    [Parameter]
+    public List<SelectEntityModel> SelectionList { get; set; } = new();
+
+    [Parameter]
+    public EventCallback<List<SelectEntityModel>?> EntityListChanged { get; set; }
+
+    [Parameter]
+    public bool Required { get; set; } = false;
+
+    #endregion
+
+    /// <summary>
+    /// Handles initial loading
+    /// </summary>
+    protected override void OnInitialized()
+    {
+        if (EntityList.Any())
+        {
+            CurrentIdList = string.Join(",", EntityList.Select(Entity => Entity.Id));
+        }
+    }
+
+    protected async Task OnIdChanged(string newValue)
+    {
+
+        CurrentIdList = newValue;
+        IEnumerable<string> selectedIdList = new List<string>();
+        if (!string.IsNullOrWhiteSpace(CurrentIdList))
+        {
+            selectedIdList = CurrentIdList.Split(',').Select(Id => Id.Trim()).ToList();
+        }
+
+        EntityList = new();
+
+        if (selectedIdList.Any())
+        {
+            foreach (string currentId in selectedIdList)
+            {
+                SelectEntityModel? entity = SelectionList.Find(Entity => 
+                !string.IsNullOrWhiteSpace(Entity.Id)
```

Comment:
the same here, actually at the moment we are not adding an entity if the id is empty...